### DISCUSSION TITLE
Fabric's 'save' Command for Obsidian on macOS

### DISCUSSION THREAD

**Discussion by cristenger** [danielmiessler/Fabric#815] [Show and tell]
For those who miss the "save" command in the new version of Fabric in macOS as a way to save outputs to an md file for Obsidian, you can use the following command that should be added to your .bashrc or .zshrc file.

```
save() {
    local FABRIC_OUTPUT_PATH="/Users/path/to/your/obsidian/folder"
    local filename="$1"

    if [[ -z "$filename" ]]; then
        echo "Usage: <comando> | save <filename>"
        echo "Example: yt --transcript URL | fabric -p PATTERN | save extract_wisdom"
        return 1
    fi

    local full_filename="${FABRIC_OUTPUT_PATH}/${filename}.md"
    
    tee -a "$full_filename"
    
    echo "Output saved in $full_filename"
}
```

**Comment by falcon997** [comment]
is it possible to save to notion with the API?

**Reply by cristenger** [reply]
It's probably possible, but Obsidian works with locally stored documents, so the solutions would be different for Notion

**Reply by Robert126** [reply]
Hi Cris,

Thank you for your great solution!

I had to change a small thing (delete a /). I am working on a Macbook.
```
save() {
    local FABRIC_OUTPUT_PATH="/Users/path/to/your/obsidian/folder"
```
to 
```
local FABRIC_OUTPUT_PATH="Users/path/to/your/obsidian/folder"
```
My installation was unable to save a file without this small change.

Kind regards,
Robert

**Comment by francisco-arturo** [comment]
thanks a lot, it worked. 
prior to this change it was possible to add a timestamp, any idea how this can be done again?
save --tag youtube-summary "obsidianNoteTitle"

this used to add timestamp + obsidiannotetitle and a tag that said youtube-summary

**Reply by productStripesAdmin** [reply]
@francisco-arturo This should work. I added a local timestamp to the script:

```
save() {
    local FABRIC_OUTPUT_PATH="/Users/path/to/your/obsidian/folder"
    local filename="$1"

    if [[ -z "$filename" ]]; then
        echo "Usage: <comando> | save <filename>"
        echo "Example: yt --transcript URL | fabric -p PATTERN | save extract_wisdom"
        return 1
    fi

    local timestamp=$(date +%Y-%m-%d)
    local full_filename="${FABRIC_OUTPUT_PATH}/${timestamp}-${filename}.md"
    
    tee -a "$full_filename"
    
    echo "Output saved in $full_filename"
}
```
